### PR TITLE
BugFix: OLM manifests 

### PR DESCRIPTION
**Pull request by annastopel**
- BugFix of issue https://github.com/kubevirt/containerized-data-importer/issues/802
          - tuned cdi-operator rbac
               1. tuned securityconstraints permission for priviledged and not
               2. added permission for get and list of storagclass resource
          - md file fixes

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #
https://github.com/kubevirt/containerized-data-importer/issues/802

**Special notes for your reviewer**:
1. Fixed API cdi provide for OLM manifests generation
2. Fixed typos in md file
3. Edited CDI operator cluster-role and CDI CR with additional attributes
**Release note**:

```release-note
Must be backported to release 1.9.0, since it is expected by hco
```

### CODE REVIEW

**Comment by awels** [comment]
ci test please

File: doc/cdi-operator-olm.md [hunk 1]
```diff
@@ -35,46 +35,64 @@ DOCKER_REPO=<repo> DOCKER_TAG=<docker tag> PULL_POLICY=<pull policy> VERBOSITY=<
...
 ```
 #### Install OLM and Marketplace operators on cluster
-- Install OLM operator from cloned operator-lifecycle-manager repo 
+- Install OLM operator from cloned operator-lifecycle-manager repo and wait untill all pods are Running and Ready. 
 
 ```bash
 kubectl apply -f $GOPATH/src/github.com/operator-framework/operator-lifecycle-manager/deploy/upstream/quickstart/olm.yaml
 ```
-- Install marketplace operator from cloned operator-marketplace repo
+- Install marketplace operator from cloned operator-marketplace repo and wait untill all pods are Running and Ready.
```
Comment (awels) [inline doc/cdi-operator-olm.md]: typo untill

File: doc/cdi-operator-olm.md [hunk 2]
```diff
@@ -35,46 +35,64 @@ DOCKER_REPO=<repo> DOCKER_TAG=<docker tag> PULL_POLICY=<pull policy> VERBOSITY=<
...
 ```bash
 kubectl apply -f $GOPATH/src/github.com/operator-framework/operator-marketplace/deploy/upstream/ --validate=false
 ```
-###CDI installation by means of OLM and Marketplace operators
+- Wait till cdi marketplace is Running and Ready.
```
Comment (awels) [inline doc/cdi-operator-olm.md]: Wait until CDI market place operator is Running and Ready

File: doc/cdi-operator-olm.md [hunk 3]
```diff
@@ -35,46 +35,64 @@ DOCKER_REPO=<repo> DOCKER_TAG=<docker tag> PULL_POLICY=<pull policy> VERBOSITY=<
...
 ```bash
-cluster/kubectl.sh apply -f _out/manifests/release/olm/operatorgroup.yaml
+kubectl apply -f _out/manifests/release/olm/operatorgroup.yaml
 ```
 - Install subscription that will point from which channel the app is downloaded
 ```bash
-cluster/kubectl.sh apply -f  _out/manifests/release/olm/cdi-subscription.yaml
+kubectl apply -f  _out/manifests/release/olm/cdi-subscription.yaml
+```
+- Verify cdi installation plan was created
```
Comment (awels) [inline doc/cdi-operator-olm.md]: Verify CDI installation plan was created
Try to keep CDI in caps as it is an abbreviation of Containerized Data Importer. I usually try to fully spell out the first CDI fully with (CDI) behind it, to give some context what CDI means.

**Comment by awels** [comment]
The CI is having issues, I saw before I restarted and after.
```namespace/cdi created
customresourcedefinition.apiextensions.k8s.io/cdis.cdi.kubevirt.io created
configmap/cdi-operator-leader-election-helper created
clusterrole.rbac.authorization.k8s.io/cdi.kubevirt.io:operator created
serviceaccount/cdi-operator created
clusterrole.rbac.authorization.k8s.io/cdi-operator-cluster created
clusterrolebinding.rbac.authorization.k8s.io/cdi-operator created
deployment.apps/cdi-operator created
cdi.cdi.kubevirt.io/cdi created
error: timed out waiting for the condition
make: *** [cluster-sync] Error 1
Took 2454 seconds
===================================
```
Its waiting for some condition that is not being satisfied.

**Comment by annastopel** [comment]
> Really hard to determine what permissions have changed since it appears that things were moved around. Can you summarize @annastopel? Also, `make cluster-sync` times out when I run locally.

@mhenriks I see that timeout too, but all cdi components are up an drunning, For now I increased the timeout to 300 seconds to see whether that helps for ci tests

**Comment by j-griffith** [comment]
The event is never getting generated:  https://gist.github.com/j-griffith/f7e979fca3f2b480a7df6f733b88f47a

File: pkg/operator/resources/operator/operator.go [hunk 4]
```diff
@@ -107,163 +107,156 @@ func getOperatorClusterRules() *[]rbacv1.PolicyRule {
...
-			},
-		},
-		{
-			APIGroups: []string{
-				"",
+				"storage.k8s.io",
 			},
 			Resources: []string{
-				"events",
+				"storageclasses",
```
Comment (j-griffith) [inline pkg/operator/resources/operator/operator.go]: Don't we still need the event resources here?

**Comment by j-griffith** [comment]
What "BugFix" does this PR address?  The commit message doesn't really help me understand the intent or why we need this.  I could probably guess from the issues list but I would be "guessing" for sure especially 6 months from now if I ever happen across this again.  Also why does it need a back port?

**Comment by mhenriks** [comment]
@annastopel I don't think increasing the timeout is going to help.  And I agree with @j-griffith need more detail about the bug and specific fix.

**Comment by annastopel** [comment]
> We seem to be missing quite a bit: https://gist.github.com/j-griffith/f7e979fca3f2b480a7df6f733b88f47a
> 
> vs pre-change:
> 
> https://gist.github.com/j-griffith/3da0bb7ba2ab4f464f3b9b5d4dac7b4d

@j-griffith  the missing parts are services we deploy for ci test - sync never gets to install them since it fails on timeout

**Comment by annastopel** [comment]
@mhenriks and @j-griffith provided below is the detailed description of the PR changes:
1. the bug:
cdi needs to vendor its api for generating the following manifests for hco csv:
- CRD
- cdi-operator rbac
- cdi-operator deployment

That api is in in https://github.com/kubevirt/containerized-data-importer/blob/master/pkg/operator/resources/operator/components.go
```
NewCdiOperatorDeployment
NewCdiOperatorClusterRole
NewCdiCrd
FillDefaults - provides default names for all cdi images
```
The bug is in FillDefaults function:
```
	if ci.APIServerImage == "" {
		ci.ClonerImage = APIServerImageDefault
	}
	if ci.UplodaProxyImage == "" {
		ci.ClonerImage = UploadProxyImageDefault
	}
	if ci.UplodaServerImage == "" {
		ci.ClonerImage = UploadServerImageDefault
	}
```
@j-griffith shall I open an issue with the detailed description on this one?

2. As part of this fix I wanted to merge between rbc cdi has and rbac hco has for cdi - to make sure we do not miss anything (operator installation vs csv installtion). Apparently, I did miss something. I am investigating it right now. Will update

3. In addition Inoticed that md file for cdi-olm has several typos and formatting errors - so decided to fix that too on this bugfix.

**Comment by annastopel** [comment]
@mhenriks  removing 
```
                       Subresources: &extv1beta1.CustomResourceSubresources{
                               Status: &extv1beta1.CustomResourceSubresourceStatus{},
                       },
```
from cdi crd resolved the issue, I do not really understand why, though

**Comment by annastopel** [comment]
> @annastopel I don't think increasing the timeout is going to help. And I agree with @j-griffith need more detail about the bug and specific fix.

It did not :) So I returned the timeout to 120sec

**Comment by annastopel** [comment]
> Really hard to determine what permissions have changed since it appears that things were moved around. Can you summarize @annastopel? Also, `make cluster-sync` times out when I run locally.

@mhenriks I organized the permissions so it will be possible to review the diff.

File: doc/cdi-operator-olm.md (hunk 1 above)
Comment (annastopel) [reply doc/cdi-operator-olm.md]: Done

File: doc/cdi-operator-olm.md (hunk 2 above)
Comment (annastopel) [reply doc/cdi-operator-olm.md]: Done

File: doc/cdi-operator-olm.md (hunk 3 above)
Comment (annastopel) [reply doc/cdi-operator-olm.md]: Done

**Comment by j-griffith** [comment]
Changes look good, update the commit message an I'll be happy to merge.

**Comment by annastopel** [comment]
ci test please
